### PR TITLE
Cli/3646 update airy cli for gcp provider

### PR DESCRIPTION
**Pull request by bilge-cakmak**
Tested by doing following

`bazel run //cli:airy`

`./bazel-bin/cli/airy create --provider gcp`  

`cd terraform/gcp-gke`
`export KUBECONFIG=../kube.conf `
`kubectl -n kube-system get service ingress-nginx-controller` gives the `EXTERNAL-IP `

### CODE REVIEW

File: cli/pkg/providers/gcp/gcp.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+func New(w io.Writer, analytics *console.AiryAnalytics) *provider {
+	return &provider{
+		w:         w,
+		analytics: *analytics,
+	}
+}
+
+func (p *provider) GetOverrides() tmpl.Variables {
+	return tmpl.Variables{
+		NgrokEnabled: true,
```
Comment (ljupcovangelski) [inline cli/pkg/providers/gcp/gcp.go:32]: This is only needed for Minikube, but not in cloud environment. This should solve the problem with ingress and the ports.

File: docs/docs/changelog.md [hunk 2]
```diff
@@ -3,6 +3,60 @@ title: Changelog
...
+- Bump preact from 10.11.0 to 10.11.2 [[#3829](https://github.com/airyhq/airy/pull/3829)]
+
+#### Airy CLI
+
+You can download the Airy CLI for your operating system from the following links:
+
+[MacOS](https://airy-core-binaries.s3.amazonaws.com/0.51.0/darwin/amd64/airy)
+[Linux](https://airy-core-binaries.s3.amazonaws.com/0.51.0/linux/amd64/airy)
+[Windows](https://airy-core-binaries.s3.amazonaws.com/0.51.0/windows/amd64/airy.exe)
+
```
Comment (ljupcovangelski) [inline docs/docs/changelog.md:59]: Not sure why the change in the changelog is here? Perhaps a rebase is needed on top of `develop`.

**Review by ljupcovangelski — CHANGES_REQUESTED** [review]
After the initial testing, everything seams to be working fine. However, the `uninstall.sh` script is not working, because the timeout for the generated `kube.conf` file expires. So we need to find a way to work around this.

File: docs/docs/getting-started/installation/gcp.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+| [**GCE**](https://cloud.google.com/kubernetes-engine/docs/concepts/node-images#cos)    | 6 Google Compute Engine Instances                                                                    | [GCE pricing calculation](https://cloud.google.com/products/calculator#id=0394acda-ce3a-4769-a453-9e2ea6ca4291)  | The instances are a part of the `Node Pool` attached to the GKE cluster. The default instance type is: `n1-standard-2`, OS type: `Container-Optimized OS`. | Yes            |
+| [**CLB**](https://cloud.google.com/load-balancing/docs/network/networklb-target-pools) | 1 Cloud Load Balancer                                                                                | [CLB pricing calculation](https://cloud.google.com/products/calculator#id=da086850-b923-475e-a4bc-9edf3d73c9ed)  | Network (target pool-based)Load Balancer created by the ingress controller Kubernetes service                                                              | No             |
+
+[^1]: Options which can be overwritten with flags to the `airy create` command.
+
+## Install
+
+To create the cluster you must setup your local GCP environment, by [configuring
```
Comment (ljupcovangelski) [inline docs/docs/getting-started/installation/gcp.md:61]: This is done in the `Configure` part, so I'm not sure we need to mention it here. If it makes more sense, you can swap the `Configure` and `Services used` sections, so that `Configure` comes right before `Install`.

File: docs/docs/getting-started/installation/gcp.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+your local GCP
+profile](https://cloud.google.com/sdk/docs/install-sdk)
+for the GCP account where all the resources will be created.
+
+Download and install the [Airy CLI](cli/introduction.md).
+
+:::warning
+
+If you want to use Airy Core with auto-generated HTTPS certificates, refer to the [Let's Encrypt section](/getting-started/installation/gcp#https-using-lets-encrypt) for customizing your `airy.yaml` file before proceeding.
```
Comment (ljupcovangelski) [inline docs/docs/getting-started/installation/gcp.md:70]: The `Let's Encrypt section` link doesn't work.

File: docs/docs/getting-started/installation/gcp.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+with the [kubectl](https://kubernetes.io/docs/tasks/tools/) command line
+utility. You can find the kubeconfig of your Airy Core instance in
+`WORKSPACE/terraform/kube.conf`.
+
+### Verification
+
+After the installation process, you can verify that all the pods are running with
+
+```
+kubectl get pods --kubeconfig .terraform/kube.conf
```
Comment (ljupcovangelski) [inline docs/docs/getting-started/installation/gcp.md:101]:
```suggestion
kubectl get pods --kubeconfig ./terraform/kube.conf
```

File: docs/docs/getting-started/installation/gcp.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+```
+
+#### Setup your DNS
```
Comment (ljupcovangelski) [inline docs/docs/getting-started/installation/gcp.md:104]: Perhaps this can be a subsection of `HTTPS using Let's Encrypt`?

File: cli/pkg/providers/gcp/kubeConfigTemplate.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+contexts:
+- context:
+    cluster: {{.ClusterName}}
+    user: {{.ClusterName}}
+  name: {{.ClusterName}}
+current-context: {{.ClusterName}}
+kind: Config
+preferences: {}
+users:
+- name: {{.ClusterName}}
```
Comment (ljupcovangelski) [inline cli/pkg/providers/gcp/kubeConfigTemplate.yaml:16]: Is this file used? I think we can remove it from Bazel, as Terraform generates the kubeconfig file.

**Review by ljupcovangelski — COMMENTED** [review]
Looking good so far. Some comments on the docs.

File: docs/docs/getting-started/installation/gcp.md (hunk 3 above)
Comment (bilge-cakmak) [reply docs/docs/getting-started/installation/gcp.md:61]: 👍

File: docs/docs/getting-started/installation/gcp.md (hunk 5 above)
Comment (bilge-cakmak) [reply docs/docs/getting-started/installation/gcp.md:96]: 👍

File: docs/docs/getting-started/installation/gcp.md (hunk 4 above)
Comment (bilge-cakmak) [reply docs/docs/getting-started/installation/gcp.md:65]: 👍 *fixed*

File: docs/docs/getting-started/installation/gcp.md (hunk 6 above)
Comment (bilge-cakmak) [reply docs/docs/getting-started/installation/gcp.md:99]: 😱 i was generously deleting things

**Review by juan-sebastian — COMMENTED** [review]
So far so good. I left some comments

File: cli/pkg/providers/gcp/gcp.go [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+	if err != nil {
+		return "", err
+	}
+	return installDir, nil
+}
+
+func (p *provider) PostInstallation(providerConfig map[string]string, namespace string, dir workspace.ConfigDir) error {
+	return nil
+}
+
```
[the remark this answers is not in the document]
Comment (bilge-cakmak) [reply cli/pkg/providers/gcp/gcp.go:61]: ohh yeah this could fly away from the provider.go interface..

File: cli/pkg/providers/gcp/gcp.go [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+	getter "github.com/hashicorp/go-getter"
+	"gopkg.in/segmentio/analytics-go.v3"
+)
+
+type provider struct {
+	w         io.Writer
+	analytics console.AiryAnalytics
+}
+
+func New(w io.Writer, analytics *console.AiryAnalytics) *provider {
```
[the remark this answers is not in the document]
Comment (bilge-cakmak) [reply cli/pkg/providers/gcp/gcp.go:24]: `Provider interface` is inn provider.go annd all our providers currently using it this way around. I think we should make another PR to adapt it in all providers' go files.

File: cli/pkg/providers/gcp/gcp.go [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+	})
+	cmd := exec.Command("/bin/bash", "install.sh")
+	cmd.Dir = installPath
+	cmd.Stdin = os.Stdin
+	cmd.Stderr = p.w
+	cmd.Stdout = p.w
+	err := cmd.Run()
+
+	if err != nil {
+		console.Exit("Error with Terraform installation", err)
```
[the remark this answers is not in the document]
Comment (bilge-cakmak) [reply cli/pkg/providers/gcp/gcp.go:84]: Hmm another design adaptation for `Provider interface` is in provider.go and all our providers currently using it this way around. I think we should make another PR.

**Review by ljupcovangelski — APPROVED** [review]
